contracts/src/subnet/SubnetActorCheckpointingFacet.sol

```diff
@@ -32,7 +32,13 @@ contract SubnetActorCheckpointingFacet is SubnetActorModifiers, ReentrancyGuard,
 
         bytes32 checkpointHash = keccak256(abi.encode(checkpoint));
 
-        if (checkpoint.blockHeight == s.lastBottomUpCheckpointHeight + s.bottomUpCheckPeriod) {
+        if (checkpoint.blockHeight <= s.lastBottomUpCheckpointHeight) {
```

Comment:
We need to add this to prevent `old` checkpoint to be resubmitted and overwrite `newer` checkpoints. This means past checkpoints cannot be submitted again.